search/views.py

```diff
@@ -1,5 +1,55 @@
 from django.shortcuts import render
+from django.contrib.auth.decorators import login_required
+from .forms import SearchForm
+from apartments.models import Apartment, City
+from datetime import datetime
 
 
+@login_required
 def search(request):
-    return render(request, 'search/search.html')
+    if request.method == 'POST':
+        search_form = SearchForm(request.POST)
+        if search_form.is_valid():
+            form = SearchForm(request.POST)
+            filtered_apartments = get_filtered_apartments(form)
+            context = {
+                'filteredApartments': filtered_apartments,
+                'loggedUser': request.user
+            }
+            return render(request, 'search/results.html', context)
+    else:
+        if request.user.is_owner is True:
+            search_form = SearchForm(
+                instance=request.user.apartment,
+                initial={
+                    'min_rent': request.user.apartment.rent - 500,
+                    'max_rent': request.user.apartment.rent + 500}
+            )
+        elif request.user.is_seeker is True:
+            search_form = SearchForm(instance=request.user.seeker)
+        else:
```

Comment:
Is there an option to not be an owner nor a seeker? Do you mean like an admin?

Reply:
Yes, that's for the admin.
@DanielMy777